api/datasets.go

```diff
@@ -73,4 +78,35 @@ func ConfigureDatasetsRouter(e *echo.Group, dldm *core.DeltaDM) {
 
 		return c.JSON(200, ads)
 	})
+
+	datasets.PUT("/:dataset_id", func(c echo.Context) error {
+		did := c.Param("dataset_id")
+		if did == "" {
+			return fmt.Errorf("dataset id not specified")
+		}
+
+		var d DatasetPutBody
+
+		if err := c.Bind(&d); err != nil {
+			return err
+		}
+
+		var existing db.Dataset
+		res := dldm.DB.Model(&db.Dataset{}).Where("id = ?", did).First(&existing)
+
+		if res.Error != nil {
+			return fmt.Errorf("error fetching dataset %s", res.Error)
+		}
+
+		if d.Name != "" {
+			existing.Name = d.Name
+		}
+
+		res = dldm.DB.Save(&existing)
+		if res.Error != nil {
+			return fmt.Errorf("error saving dataset %s", res.Error)
+		}
+
+		return c.JSON(http.StatusOK, existing)
```

Comment:
Good thinking here with `http.StatusOK`. Would you mind making all our other API responses consistent with this too? I think some of them vary - some are a hardcoded int and some have the official variable.

Reply:
I went through all files and changed it to use `http.StatusOK` to all hardcoded values that I found.